dask/dataframe/io.py

```diff
@@ -100,7 +100,7 @@ def read_csv(fn, *args, **kwargs):
 
     if 'nrows' in kwargs:  # Just create single partition
         dsk = {(name, 0): (apply, pd.read_csv, (fn,),
-                                  assoc(kwargs, 'header', header))},
```

Comment:
It was, but through some other special casing not meant to catch this, it ended up working anyway.